hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/footer/DeleteActionsLayout.java

```diff
@@ -258,46 +257,41 @@ private void deleteTargetTag(final Component source) {
     }
 
     private void addInDeleteDistributionList(final Table sourceTable, final TableTransferable transferable) {
-        final Set<DistributionSetIdName> distSelected = AbstractTable.getTableValue(sourceTable);
-        final Set<DistributionSetIdName> distributionIdNameSet = new HashSet<>();
-
-        if (!distSelected.contains(transferable.getData(SPUIDefinitions.ITEMID))) {
-            distributionIdNameSet.add((DistributionSetIdName) transferable.getData(SPUIDefinitions.ITEMID));
-        } else {
-            distributionIdNameSet.addAll(distSelected);
-        }
+        @SuppressWarnings("unchecked")
+        final AbstractTable<?, DistributionSetIdName> distTable = (AbstractTable<?, DistributionSetIdName>) sourceTable;
+        final Set<DistributionSetIdName> distributionIdNameSet = distTable.getDeletedEntityByTransferable(transferable);
 
         final DistributionSetIdName dsInBulkUpload = managementUIState.getTargetTableFilters().getBulkUpload()
                 .getDsNameAndVersion();
         if (isDsInUseInBulkUpload(distributionIdNameSet, dsInBulkUpload)) {
             distributionIdNameSet.remove(dsInBulkUpload);
         }
 
-        if (!distributionIdNameSet.isEmpty()) {
-
+        if (distributionIdNameSet.isEmpty()) {
+            return;
+        }
+        /*
+         * Flags to identify whether all dropped distributions are already in
+         * the deleted list (or) some distributions are already in the deleted
+         * distribution list.
+         */
+        final int existingDeletedDistributionsSize = managementUIState.getDeletedDistributionList().size();
+        managementUIState.getDeletedDistributionList().addAll(distributionIdNameSet);
+        final int newDeletedDistributionsSize = managementUIState.getDeletedDistributionList().size();
+        if (newDeletedDistributionsSize == existingDeletedDistributionsSize) {
```

Comment:
Good question. Maybe in a first step you can simplify it?
Try: `if (distributionIdNameSet.isEmpty())`

I suppose that `managementUIState.getDeletedDistributionList()` returns a list, so `newDeletedDistributionsSize == existingDeletedDistributionsSize` can only be true if `distributionIdNameSet` don't contain anything, right?

Can you check the UI error message? Does it fit to that assumption?